src/bosh-google-cpi/google/disk_service/google_disk_service_resize.go

```diff
@@ -0,0 +1,42 @@
+package disk
+
+import (
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+
+	"bosh-google-cpi/api"
+	"bosh-google-cpi/util"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func (d GoogleDiskService) Resize(id string, newSize int) error {
+	newsizeGB := &compute.DisksResizeRequest{
+		SizeGb: int64(newSize),
+	}
+
+	disk, found, err := d.Find(id, "")
+	if err != nil {
+		return err
+	}
+	if !found {
+		return api.NewDiskNotFoundError(id, false)
+	}
+
+	if disk.SizeGb == newsizeGB.SizeGb {
+		d.logger.Debug(googleDiskServiceLogTag, "Skipping resize Google Disk '%s', becasue current value '%s'is equal to new value '%s'", id, disk.SizeGb, newsizeGB)
+	} else if disk.SizeGb > newsizeGB.SizeGb {
+		return bosherr.WrapErrorf(err, "Skipping resize Google Disk '%s', cannot resize volume to a smaller size from '%s' to '%s'", id, disk.SizeGb, newsizeGB)
```

Comment:
I get this error when running `make test` in `src/bosh-google-cpi` directory:
```
google/disk_service/google_disk_service_resize.go:28:10: WrapErrorf format %s has arg disk.SizeGb of wrong type int64
```

Reply:
fixed in 3f36de27258ead42ef9af12faf7296125fb1b8c9